test/conformance/ingress/httpoption.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"knative.dev/networking/pkg/apis/networking"
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/networking/test"
+)
+
+// TestHTTPOption verifies that the Ingress properly handles HTTPOption field.
+func TestHTTPOption(t *testing.T) {
+	// net-istio cannot support parallel option as one HTTPOption effects globally.
+	// https://github.com/knative-sandbox/net-istio/issues/637
+	// t.Parallel()
+	ctx, clients := context.Background(), test.Setup(t)
+
+	tests := []struct {
+		name       string
+		httpOption v1alpha1.HTTPOption
+		code       int
+	}{{
+		name:       "enabled",
+		httpOption: v1alpha1.HTTPOptionEnabled,
+		code:       http.StatusOK,
+	}, {
+		name:       "redirected",
+		httpOption: v1alpha1.HTTPOptionRedirected,
+		code:       http.StatusMovedPermanently,
+	}}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// net-istio cannot support parallel option as one HTTPOption effects globally.
+			// https://github.com/knative-sandbox/net-istio/issues/637
```

Comment:
Checking my understanding here: istio supports this, but it flips the setting for all ingresses back and forth each time it receives one? If so does that mean running this test will affect the whole cluster it runs against temporarily (because I think some people might run the tests against production clusters to test after deploying, so that may not be good..)?

Reply:
> istio supports this, but it flips the setting for all ingresses back and forth each time it receives one?

Yes, exactly. Istio's HTTP is global. I created the issue https://github.com/knative-sandbox/net-istio/issues/637

> If so does that mean running this test will affect the whole cluster it runs against temporarily (because I think some people might run the tests against production clusters to test after deploying, so that may not be good..)?

Yes, that's also true. Perhaps we should NOT comment out this and let net-istio fix the issue  https://github.com/knative-sandbox/net-istio/issues/637.

Reply:
yeh, I think we may need to fix net-istio before landing this, unfortunately

Reply:
And this conformance test should add an extra test that having multiple Ingress with different HTTPOption at the same time.